src/Lti/Factory/Lti1p1RequestFactory.php

```diff
@@ -0,0 +1,157 @@
+
+    /** @var NonceGenerator */
+    private $generator;
+
+    /** @var RouterInterface */
+    private $router;
+
+    /** @var LtiInstanceLoadBalancerInterface */
+    private $loadBalancer;
+
+    /** @var string */
+    private $ltiLaunchPresentationReturnUrl;
+
+    /** @var string */
+    private $ltiLaunchPresentationLocale;
+
+    public function __construct(
+        OAuthSigner $signer,
+        NonceGenerator $generator,
+        RouterInterface $router,
+        LtiInstanceLoadBalancerInterface $loadBalancer,
+        string $ltiLaunchPresentationReturnUrl,
+        string $ltiLaunchPresentationLocale
+    ) {
+        $this->signer = $signer;
+        $this->generator = $generator;
+        $this->router = $router;
+        $this->loadBalancer = $loadBalancer;
+        $this->ltiLaunchPresentationReturnUrl = $ltiLaunchPresentationReturnUrl;
+        $this->ltiLaunchPresentationLocale = $ltiLaunchPresentationLocale;
+    }
+
+    /**
+     * @throws AssignmentNotProcessableException
+     * @throws JsonException
+     */
+    public function create(Assignment $assignment): LtiRequest
+    {
+        $ltiInstance = $this->loadBalancer->getLtiInstance($assignment->getUser());
+
+        $context = new OAuthContext(
+            '',
+            $ltiInstance->getLtiKey(),
+            $this->generator->generate(),
+            OAuthContext::METHOD_MAC_SHA1,
+            (string)Carbon::now()->getTimestamp(),
+            OAuthContext::VERSION_1_0
+        );
+
+        $ltiLink = $this->getLtiLaunchLink($ltiInstance, $assignment->getLineItem());
+        $ltiParameters = $this->getAssignmentLtiParameters($assignment);
+
+        $signature = $this->signer->sign(
+            $context,
+            $ltiLink,
+            Request::METHOD_POST,
+            $ltiInstance->getLtiSecret(),
+            $ltiParameters
+        );
+
+        return new LtiRequest(
+            $ltiLink,
+            LtiRequest::LTI_VERSION_1P0,
+            array_merge(
+                [
+                    'oauth_body_hash' => $context->getBodyHash(),
+                    'oauth_consumer_key' => $context->getConsumerKey(),
+                    'oauth_nonce' => $context->getNonce(),
+                    'oauth_signature' => $signature,
+                    'oauth_signature_method' => $context->getSignatureMethod(),
+                    'oauth_timestamp' => $context->getTimestamp(),
+                    'oauth_version' => $context->getVersion(),
+                ],
+                $ltiParameters
+            )
+        );
+    }
+
+    private function getLtiLaunchLink(LtiInstance $ltiInstance, LineItem $lineItem): string
+    {
+        return sprintf(
+            '%s/ltiDeliveryProvider/DeliveryTool/launch/%s',
+            rtrim($ltiInstance->getLtiLink(), '/'),
+            base64_encode(
+                json_encode(
+                    ['delivery' => $lineItem->getUri()],
+                    JSON_THROW_ON_ERROR,
+                    512
```

Comment:
Just a check, why do we need to set `depth = 512`, since `512` is the default value?

Reply:
It's not exactly mine, but removed :)